server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -6251,9 +6267,115 @@ public UserVm createVirtualMachine(DeployVMCmd cmd) throws InsufficientCapacityE
                 }
             }
         }
+
+        applyLeaseOnCreateInstance(vm, leaseDuration, leaseExpiryAction, svcOffering);
         return vm;
     }
 
+    protected void validateLeaseProperties(Long leaseDuration, String leaseExpiryAction) {
+        if (!VMLeaseManagerImpl.InstanceLeaseEnabled.value()
+                || (leaseDuration == null && StringUtils.isEmpty(leaseExpiryAction))) { // if both are null
+            return;
+        }
+
+        boolean bothValuesSet = true;
+        if (leaseDuration != null) {
+            if (leaseDuration == -1) {   // special condition used to disable lease for instance
+                return;
+            }
+            // both params have value
+            if (leaseDuration < -1) {
```

Comment:
what if `leaseDuration` is 0, better consider zero or less than zero as lease disabled (update the cmd description if required).